tutorials/packaging/snap-a-website.md

```diff
@@ -0,0 +1,670 @@
+---
+id: snap-a-website
+summary: In this tutorial, you’ll learn how to create a desktop web app with Electron and package it as a snap.
+categories: packaging
+tags: snapcraft, snap, beginner, nodejs, electron, website
+status: Published
+feedback Link: https://github.com/ubuntu/codelabs/issues:
+author: David Callé <david.calle@canonical.com>
+difficulty: 2
+published: 2017-11-14
+
+---
+
+# Snap a website with Electron
+
+## Overview
+Duration: 1:00
+
+Turning your website into a desktop integrated app is a relatively simple thing to do, but distributing it as such and making it noticeable in app stores is another story. This tutorial will show you how to leverage Electron and snaps to create a website-based desktop app from scratch and publish it on a multi-million users app store shared between many Linux distributions.
+
+For this tutorial, the website we are going to package is an HTML5 game called [Castle Arena](http://castlearena.io).
+
+### What you'll learn
+
+- How to create a website-based desktop app using Electron
+- How to turn it into a snap package
+- How to make your first release in the Snap Store
+
+### What you'll need
+
+- Ubuntu Desktop 16.04 or above
+- Some basic command-line knowledge
+
+We are going to start by downloading some dev tools and set up the project.
+
+## Electron setup
+Duration: 10:00
+
+Let's start by opening a terminal and installing some basic development tools, notably:
+
+* **curl**: a command-line utility to download remote content
+* **build-essential**: a collection of dependencies for building source code
+* **nodejs**: A JavaScript runtime, that will also provide the `npm` command we are going to use
+
+```bash
+sudo apt install curl build-essential libgconf2-4
+curl -sL https://deb.nodesource.com/setup_9.x | sudo -E bash -
+sudo apt install nodejs
+```
+
+Create a `castlearena` project directory and enter it:
+
+```bash
+mkdir castlearena
+cd castlearena
+```
+
+Inside our project, let's create an `app` directory to host the Electron app and its dependencies, and enter it.
+
+```bash
+mkdir app
+cd app
+```
+
+With the `npm` command, we can install electron and electron-builder inside the current directory. They will provide the framework and tooling to build our app binary.
```

Comment:
I get that 'electron' here is used as the name of the package - maybe it would make more sense to italicise these?